### ISSUE TITLE
Error: Undefined constant in HttpClientAdapterGuzzle.php:106

### ISSUE THREAD

**Issue by Pantos** [mstilkerich/carddavclient#20]
I installed Roundcube 1.6 today. After that I ran `composer update --no-dev`. First I ran into the issue  mstilkerich/rcmcarddav#388. I removed the plugin and installed it again. The plugin seems to working now.

But If I want to look into an address I get an `500 Internal Server Error` now. The Roundcube error.log shows:

```
30-Jul-2022 18:36:00 Europe/Berlin] PHP Fatal error:  Uncaught Error: Undefined constant "MStilkerich\CardDavClient\CURLAUTH_NEGOTIATE" in /html/vendor/mstilkerich/carddavclient/src/HttpClientAdapterGuzzle.php:106
Stack trace:
#0 /html/vendor/mstilkerich/carddavclient/src/CardDavClient.php(58): MStilkerich\CardDavClient\HttpClientAdapterGuzzle->__construct('https://cloud.tes...', Array)
mstilkerich/rcmcarddav#1 /html/vendor/mstilkerich/carddavclient/src/Account.php(143): MStilkerich\CardDavClient\CardDavClient->__construct('https://cloud.tes...', Array)
mstilkerich/rcmcarddav#2 /html/vendor/mstilkerich/carddavclient/src/Account.php(216): MStilkerich\CardDavClient\Account->getClient()
mstilkerich/rcmcarddav#3 /html/vendor/mstilkerich/carddavclient/src/Services/Discovery.php(120): MStilkerich\CardDavClient\Account->findCurrentUserPrincipal('/.well-known/ca...')
mstilkerich/rcmcarddav#4 /html/plugins/carddav/carddav.php(1507): MStilkerich\CardDavClient\Services\Discovery->discoverAddressbooks(Object(MStilkerich\CardDavClient\Account))
mstilkerich/rcmcarddav#5 /html/plugins/carddav/carddav.php(630): carddav->determineAddressbooksToAdd(Object(MStilkerich\CardDavClient\Account))
mstilkerich/rcmcarddav#6 /html/program/lib/Roundcube/rcube_plugin_api.php(518): carddav->savePreferences(Array)
mstilkerich/rcmcarddav#7 /html/program/actions/settings/prefs_save.php(165): rcube_plugin_api->exec_hook('preferences_sav...', Array)
mstilkerich/rcmcarddav#8 /html/program/include/rcmail.php(275): rcmail_action_settings_prefs_save->run(Array)
mstilkerich/rcmcarddav#9 /html/index.php(278): rcmail->action_handler()
mstilkerich/rcmcarddav#10 {main}
  thrown in /html/vendor/mstilkerich/carddavclient/src/HttpClientAdapterGuzzle.php on line 106
  thrown in /html/vendor/mstilkerich/carddavclient/src/HttpClientAdapterGuzzle.php on line 106
```
Has someone else the same problem? Is it a general bug or an individual problem? Any idea how to solve it?

**Comment by mstilkerich** [comment]
Hello, I created a test account with your provider and think I figured out the issue. The curl version is compiled without support for SPNEGO, and this seems to result in that constant not being defined. I was not aware that the availability of the constant was bound to the feature being compiled in. I have released v1.2.2 of carddavclient which should fix the issue. If you run `composer update --no-dev`, you should get that version.

Watch out for:
```
  - Upgrading mstilkerich/carddavclient (v1.2.1 => v1.2.2)
```

**Comment by Pantos** [comment]
@mstilkerich: Thanks a lot for your very fast reaction. It works fine now!
